### PR TITLE
fix: 使API余额查询可用

### PR DESCRIPTION
**Pull request by luckywangxi**
image

### CODE REVIEW

File: service/src/chatgpt/index.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export async function initApi() {
     const options: ChatGPTAPIOptions = {
       apiKey: config.apiKey,
       completionParams: { model },
-      debug: !config.apiDisableDebug,
+      debug: true,
```
Comment (Kerwin1202) [inline service/src/chatgpt/index.ts:46]: 这个是刚改的... 被覆盖了

File: service/src/chatgpt/index.ts [hunk 2]
```diff
@@ -122,28 +122,68 @@ async function chatReplyProcess(options: RequestOptions) {
...
+    response = await fetch(urlUsage, {headers});
+    const usageData = await response.json();
+    const totalUsage = usageData.total_usage / 100;
+
+    // 计算剩余额度
+    const balance = totalAmount - totalUsage;
+
+    // 输出余额信息
+    console.log(`balance: ${balance.toFixed(3)}`);
+
```
Comment (Kerwin1202) [inline service/src/chatgpt/index.ts:171]: 这种console.log不删掉没办法 编译 不过去检验

File: service/src/chatgpt/index.ts [hunk 3]
```diff
@@ -155,20 +195,17 @@ async function chatConfig() {
 
 async function setupProxy(options: ChatGPTAPIOptions | ChatGPTUnofficialProxyAPIOptions) {
   const config = await getCacheConfig()
-  if (isNotEmptyString(config.socksProxy)) {
+  if (config.socksProxy) {
```
Comment (Kerwin1202) [inline service/src/chatgpt/index.ts:198]: 这种就是冲突被覆盖的

**Review by Kerwin1202 — COMMENTED** [review]
一些问题

**Comment by cmtomkan310** [comment]
大佬，刚刚docker部署了一下你的源，出问题

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

> chatgpt-web-service@1.0.0 prod /app
> esno ./build/index.js

/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9
    return (connectionString.startsWith('mongodb://') ||
                             ^


TypeError: Cannot read properties of undefined (reading 'startsWith')
    at connectionStringHasValidScheme (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:9:30)
    at ConnectionString (/app/node_modules/.pnpm/mongodb-connection-string-url@2.6.0/node_modules/mongodb-connection-string-url/lib/index.js:85:34)
    at parseOptions (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/connection_string.js:204:17)
    at MongoClient (/app/node_modules/.pnpm/mongodb@5.1.0/node_modules/mongodb/lib/mongo_client.js:46:50)
    at __defProp (/app/build/index.js:9545:14)
    at Object. (/app/build/index.js:10399:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.F (/app/node_modules/.pnpm/@esbuild-kit+cjs-loader@2.4.2/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:941)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.

**Comment by Kerwin1202** [comment]
@cmtomkan310 代表没找到 环境变量 你看下跑的目录下是否有 .env 文件 原来是一个 .env.example 需要改内容 并重命名

**Comment by Jxells** [comment]
SITE_TITLE 设置了不生效

**Comment by Kerwin1202** [comment]
> SITE_TITLE 设置了不生效

docker 可以的  自己编译是不行，因为是docker启动的时候替换的，自己编译 只是在发邮件 里的内容会变 html不会 这种自己编译就自己手动改 index.html 吧，我这目前侧重的是docker 跑的

**Comment by Jxells** [comment]
> > SITE_TITLE 设置了不生效
> 
> docker 可以的 自己编译是不行，因为是docker启动的时候替换的，自己编译 只是在发邮件 里的内容会变 html不会 这种自己编译就自己手动改 index.html 吧，我这目前侧重的是docker 跑的

ok 感谢

**Comment by Jxells** [comment]
是否考虑增加每个用户可以自定义设置自己的OPENAI_API_KEY或者OPENAI_ACCESS_TOKEN

**Comment by luckywangxi** [comment]
> 我合并了下但是发现冲突以及一些问题有点多， 所以直接自己抄你代码提交了。。

谢谢大佬🌚

**Comment by Kerwin1202** [comment]
> > 我合并了下但是发现冲突以及一些问题有点多， 所以直接自己抄你代码提交了。。
> 
> 谢谢大佬🌚

应该是你fork https://github.com/Chanzhaoyu/chatgpt-web 这个改导致的，，你fork我的改就不会，，现在每次我合并分支要改一堆冲突。。

**Comment by jtwgo** [comment]
为什么我部署完成后，注册了一个邮箱，通过邮箱的登录后，点击设置，只出现一个总览，没有其他配置的选项呢

**Comment by Kerwin1202** [comment]
> 为什么我部署完成后，注册了一个邮箱，通过邮箱的登录后，点击设置，只出现一个总览，没有其他配置的选项呢

设置管理邮箱  环境变量 `ROOT_USER` 设置为第一个邮箱 才会显示

**Pull request closed without merging**